### ISSUE TITLE
Wrong handling TRUE, FALSE and NULL? Should be lowercase?

### ISSUE THREAD

**Issue by Daijobou** [koseven/koseven#226]
With kohana 3 I change my code style to uppercase, but this is wrong?

PHP accept both, but look to other code languages, lowercase is a MUST.

Javascript is case-sensitive, so only allow lowercase boolean and null.

json said https://tools.ietf.org/html/rfc7159#section-3: 
> The literal names MUST be lowercase.  No other literal names are allowed.

PHP json_decode said (source https://secure.php.net/manual/en/migration56.incompatible.php)
> json_decode() now rejects non-lowercase variants of the JSON literals true, false and null at all times, as per the JSON specification, and sets json_last_error() accordingly.

Uppercase bools are constants and lowercases are values.

https://www.php-fig.org/psr/psr-2/
> The PHP constants true, false, and null MUST be in lower case.

BUT

> To specify a boolean literal, use the constants TRUE or FALSE. Both are case-insensitive.
Source: http://php.net/manual/en/language.types.boolean.php

EDIT:
I take a short look in other frameworks

laraval = lowercase
cphalcon = lowercase
codeigniter = **uppercase**
symfony = lowercase
cakephp = lowercase
zend = lowercase
fullphp = lowercase
slim = lowercase

Conclusion: I found only kohana and codeigniter to use uppercase.

**Comment by neo22s** [comment]
It's the coding standard we always had.

For me it's fine I like it as it is.

https://docs.koseven.ga/guide/kohana/conventions

**Comment by saruman** [comment]
I like it as it is as well.

**Comment by Daijobou** [comment]
@bato3 
Your comment to C++ is not 100% _right_. In C++ true is a boolean and TRUE is a integer constant (1). FALSE is not defined by the C++ standard. Its defined with preprocessor macros and depends on the platform.

In SQL you should write anything command/function uppercase that not a table or field or value, for better reading a "String". I don't think that have anything todo with this discuss here.

I not say koseven should go strict with PSR-2. I quote only what I found to this topic :)
Its fine for me too. At last, I'm not forced to stick to it. So everyone can ignore the koseven conventions, when he want use lowercase.

The main reason I wrote this topic is, that its more useful to use lowercase in koseven conventions, because most languages use lowercase too and you must not switch your code style between this languages. In particular between Javascript <-> PHP <-> JSON

**Comment by bato3** [comment]
Where I wrote, that C++ use upper case? (i wrote: NULL, true,  false)

And where I wrote, that SQL need uppercase? (Convention != require)

And I wrote, that **partial standards implementation is wrong!**

_Arrgh! This library don't use PSR-2. I can use it..._ You must use "wrong" convention only when you want pull changes to koseven "core". And should use if you want create ko7 external module. Finally: your project - your rules...

PS.: JSON? You manually create json? So this should be JS <-> PHP ( <-> SQL)

BTW; I'm wrong, or python require True/False?
